pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -317,3 +408,22 @@ func newIdentity(userAcc *toolchainv1alpha1.UserAccount, user *userv1.User) *use
 func ToIdentityName(userID string) string {
 	return fmt.Sprintf("%s:%s", config.GetIdP(), userID)
 }
+
+func containsString(items []string, value string) bool {
+	for _, item := range items {
+		if item == value {
+			return true
+		}
+	}
+	return false
+}
+
+func removeString(items []string, value string) (result []string) {
+	for _, item := range items {
```

Comment:
minor suggestion: you could probably initialize the `result` array with a capacity of `len(items) - 1`